### ISSUE TITLE
ptr_arg lint could warn also with &PathBuf and &Path

### ISSUE THREAD

**Issue by MarcoIeni** [rust-lang/rust-clippy#6502]
### What it does

The [ptr_arg](https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg) lint already warns you when you pass a `&String` to a function.

For example with this piece of code:

```rust
fn print(a: &String) {
    println!("{:?}", a);
}

fn main() {
    let a = "/home/marco".to_string();
    print(&a);
}
```

This warning is emitted:

![image](https://user-images.githubusercontent.com/11428655/103074328-aa218a80-45c9-11eb-9abe-5385c520d9a2.png)

It would be nice if it did the same for `PathBuf` and `Path`, because from my understanding, the same principle of `String` and `str` applies.
### Example

```rust
use std::path::PathBuf;

fn print(a: &PathBuf) {
    println!("{:?}", a);
}

fn main() {
    let a: PathBuf = "/home/marco".into();
    print(&a);
}
```

Could be written as:

```rust
use std::path::{Path, PathBuf};

fn print(a: &Path) {
    println!("{:?}", a);
}

fn main() {
    let a: PathBuf = "/home/marco".into();
    print(&a);
}
```
